react-carousel/my-app/src/Carousel.tsx

```diff
@@ -0,0 +1,86 @@
+import {
+  FaChevronCircleLeft,
+  FaChevronCircleRight,
+  FaCircle,
+  FaRegCircle,
+} from 'react-icons/fa';
+import { Buttons } from './Buttons';
+import { ImageDisplay } from './ImageDisplay';
+import { useEffect, useState } from 'react';
+
+export function Carousel() {
+  const [imageIndex, setImageIndex] = useState(0);
+  const images = [
+    {
+      src: '/images/fushiguro.webp',
+      alt: 'Megumi Fushiguro',
+    },
+    {
+      src: '/images/inumaki.webp',
+      alt: 'Toge Inumaki',
+    },
+    {
+      src: '/images/itadori.webp',
+      alt: 'Yuji Itadori',
+    },
+    {
+      src: '/images/kugisaki.webp',
+      alt: 'Nobara Kugisaki',
+    },
+    {
+      src: '/images/panda.webp',
+      alt: 'Panda',
+    },
+    {
+      src: '/images/zen-in.webp',
+      alt: "Maki Zen'in",
+    },
+  ];
+  // const callback = useCallback(() => imageIndex, []);
+  useEffect(() => {
+    const intervalId = setTimeout(
+      () => setImageIndex((prev) => (prev >= images.length - 1 ? 0 : prev + 1)),
+      1000
+    );
+    return () => clearTimeout(intervalId);
+  }, [imageIndex]);
+  const selectorButtons = images.map((_, index) => {
+    return (
+      <Buttons
+        key={index}
+        onClick={() => handleSelect(index)}
+        appearance={imageIndex === index ? <FaCircle /> : <FaRegCircle />}
+      />
+    );
+  });
+  function handleNext() {
```

Comment:
This is the same thing you are doing on line 42 inside the setTimeout arrow function.

To avoid the code duplication you can make use of `useCallback`. Turn this into a `useCallback` function (and move it up in front of the `useEffect`):

`const handleNext = () => { ... };`

then pass `handleNext` to the `setTimeout` call.